frontend/packages/ceph-storage-plugin/integration-tests/tests/2-tests/add-capacity.scenario.ts

```diff
@@ -105,63 +84,177 @@ if (clusterStatus && cephHealth) {
-      await click(storageClusterLink);
+      await selectSCDropdown(expansionObjects.uid);
 
-      await clickKebabAction(uid, 'Add Capacity');
-      await verifyFields();
+      // eslint-disable-next-line no-useless-escape
+      expansionObjects.defaultSC = execSync(
+        `kubectl get storageclasses | grep -Po '\\w+(?=.*default)'`,
+      )
+        .toString()
+        .trim();
+      await click(getSCOption(expansionObjects.defaultSC));
+      verifyFields();
       await click(confirmButton);
 
-      const statusCol = storageClusterRow(uid).$('td:nth-child(3)');
-
-      // need to wait as cluster states fluctuates for some time. Waiting for 2 secs for the same
-      await browser.sleep(2 * SECOND);
-
-      await browser.wait(until.textToBePresentInElement(statusCol, CLUSTER_STATUS.PROGRESSING));
-      await browser.wait(
-        until.textToBePresentInElement(
-          statusCol.$('span.co-icon-and-text span'),
-          CLUSTER_STATUS.READY,
-        ),
-      );
+      await browser.sleep(5 * SECOND);
 
       expansionObjects.updatedClusterJSON = JSON.parse(
-        execSync(`kubectl get -o json -n ${NS} ${KIND} ${name}`).toString(),
+        execSync(`kubectl get -o json -n ${NS} ${KIND} ${expansionObjects.name}`).toString(),
       );
       expansionObjects.updatedCnt = _.get(
         expansionObjects.updatedClusterJSON,
         'spec.storageDeviceSets[0].count',
       );
-      expansionObjects.updatedPods = JSON.parse(
-        execSync(`kubectl get pod -o json -n ${NS}`).toString(),
-      );
+    });
 
-      // need to wait to get the new osds reflected in osd tree
-      await browser.sleep(1 * MINUTE);
+    it('Newly added capacity should takes into effect at the storage level', () => {
+      // by default 2Tib capacity is being added
+      expect(expansionObjects.updatedCnt - expansionObjects.previousCnt).toEqual(1);
+    });
 
-      expansionObjects.updatedOSDTree = JSON.parse(
-        execSync(
-          `oc -n ${NS} rsh  $(oc -n ${NS} get pod | grep ceph-operator| awk '{print$1}') ceph --conf=/var/lib/rook/${NS}/${NS}.config osd tree --format=json`,
-        ).toString(),
+    it('Selected storage class should be sent in the YAML', () => {
+      const storageCR = JSON.parse(
+        execSync(`kubectl get storageclusters ${STORAGE_CLUSTER_NAME} -n ${NS} -o json`).toString(),
       );
+      const scFromYAML =
+        storageCR?.spec?.storageDeviceSets?.[0]?.dataPVCTemplate?.spec?.storageClassName;
+      expect(expansionObjects.defaultSC).toEqual(scFromYAML);
+    });
+  });
 
-      expansionObjects.formattedOSDTree = createOSDTreeMap(
-        expansionObjects.updatedOSDTree.nodes,
-      ) as FormattedOsdTreeType;
-      expansionObjects.newOSDIds = getNewOSDIds(
-        expansionObjects.updatedOSDTree.nodes,
-        expansionObjects.previousOSDIds,
-      );
-    }, EXPAND_WAIT);
+  describe('Addition tests for Baremetal infra', () => {
```

Comment:
Expansion tests should run all the same even for Baremetal Expansion. This whole test should be configurable to accept a storage class via the environment variable.

Reply:
This PR targets the ocp ci tests.  These changes can be done as a part of OCS E2E tests.